docs/proposals/07-shoot-control-plane-migration.md

```diff
@@ -201,44 +201,117 @@ The ShootState synchronization controller will become part of the gardenlet. It
 
-Horizontal Pod Autoscalers also need to be considered and removed.
+```go
+type CopyOperation struct {
+	// Status is the current status of the copy operation, one of "Initial", "Ready", or "Done".
+	Status OperationStatus `json:"status"`
+}
+```
+
+The mechanism to copy the snapshots and pass the ownership from the source to the destination seed consists of the following steps:
+
+1. The reconciliation flow ("restore" phase) is triggered in the destination seed without first executing the migration flow in the source seed (or perhaps it was executed, but it failed, and its state is currently unknown).
+
+2. An additional "source" backup entry referencing the source seed backup bucket is deployed to the Garden cluster and the destination seed and reconciled by the backup entry controller. As a result, a secret with the appropriate credentials for accessing the source seed backup container named `source-etcd-backup` is created in the destination seed. The normal backup entry (referencing the destination seed backup container) is also deployed and reconciled, as usual, resulting in the usual `etcd-backup` secret being created.
+
+3. A special "copy" version of the `etcd-main` Etcd resource is deployed to the destination seed. In its `backup` section, this resource contains a `sourceStore` in addition to the usual `store`, which contains the parameters needed to use the source seed backup container, such as its name and the secret created in the previous step.
+
+   ```yaml
+   spec:
+     backup:
+       ...
+       store:
+         container: 408740b8-6491-415e-98e6-76e92e5956ac
+         secretRef:
+           name: etcd-backup
+         ...
+       sourceStore:
+         container: d1435fea-cd5e-4d5b-a198-81f4025454ff
+         secretRef:
+           name: source-etcd-backup
+         ...
+   ```
+
+4. The `etcd-druid` in the destination seed reconciles the above resource by deploying a `etcd-main` StatefulSet that doesn't contain the main `etcd` container at all. Instead, it only contains the `backup-restore` container that runs `etcd-backup-restore` configured to run in "copy" mode, which would only perform copying of the snapshots from the source to the destination backup container.
+   
+5. Before starting the copy itself, the `etcd-backup-restore` process in the destination seed creates the `CopyOperation` sync object mentioned above (if it doesn't already exist; see also [Migration Flow Adaptations](#migration-flow-adaptations)) with its status set to `Initial`. It then waits for the status of this sync object to become `Ready` in order to proceed. This waiting is up to a certain timeout that should be sufficient for a full snapshot to be taken; after this timeout has elapsed, it sets the status to `Ready` on its own, and the reconciliation flow continues from step 8, as it relies on the safety mechanism described in [Implicit Owner Election](#implicit-owner-election) below.
```

Comment:
> the `etcd-backup-restore` process in the destination seed creates the `CopyOperation` sync object mentioned above

Where is this created?

Reply:
The `etcd-backup-restore` process is started in the `backup-restore` container of the special `etcd-main` stateful set mentioned in step 4, just above:
> Instead, it only contains the `backup-restore` container that runs `etcd-backup-restore` configured to run in "copy" mode

Or is your question about something else?

Reply:
Sorry, I meant where the `CopyOperation` sync object is created - I assume in the backup bucket/entry itself?

Reply:
Yes, in the backup bucket, as a kind of object different from "snapshot".